server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MetadataRolloverService.java

```diff
@@ -84,6 +84,7 @@ public class MetadataRolloverService {
         AutoShardingType.COOLDOWN_PREVENTED_DECREASE,
         "es.auto_sharding.cooldown_prevented_decrease.total"
     );
+    private static final String NON_EXISTENT_SOURCE = "non-existent";
```

Comment:
Ah I didn't consider the fact that `non-existent` could also be an index name. I think making `RolloverResponse.oldIndex` nullable is a good solution. Wouldn't it make more sense to just use the `TransportVersion` to change the (de)serialization, though?

Reply:
I think I prefer a null value for a previously nonexistent index. Trying to think of the edge cases for it though. The main thing that comes to mind is that an unconditional rollover operation on the data stream targeting the failure store when there are no failure indices would likely result in a response to the client that has a null value for the old index field. We will probably need to make sure that the client libraries will be able to accommodate such a change, since I believe right now it's not possible for it to be null.

Reply:
+1 on checking this. The good thing for this, is that this applied only to failure store, so a change in the client would be necessary to begin with, so maybe we can see if this is enough to make it nullable. Should we contact the clients team to begin with?

Reply:
Making this `null` will indeed break all the older clients that don't make changes (I spoke with Sylvain about this). We guarantee that 8.14.0 clients will work with all the 8.14+ releases, and this would break that for strongly-typed clients unfortunately. I think either an empty string on a placeholder like `_none_` that is never a valid index name would be better.

Reply:
++ to `_none_` instead of empty string for the time being